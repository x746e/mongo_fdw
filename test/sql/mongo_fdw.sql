CREATE SERVER mongo_server FOREIGN DATA WRAPPER mongo_fdw OPTIONS (address '127.0.0.1', port '27017');

\! mongoimport --db mongo_fdw_regress --collection countries --jsonArray --drop --quiet < test/sql/mongo_fixture.json

--
-- Basic types
CREATE FOREIGN TABLE countries (
    _id NAME,
    name VARCHAR,
    population INTEGER,
    capital VARCHAR,
    hdi FLOAT
) SERVER mongo_server OPTIONS (database 'mongo_fdw_regress', collection 'countries');
SELECT * FROM countries;

--
-- Subfields and dates
CREATE FOREIGN TABLE country_elections (
    _id NAME,
    "lastElections.type" VARCHAR,
    "lastElections.date" TIMESTAMP
) SERVER mongo_server OPTIONS (database 'mongo_fdw_regress', collection 'countries');
SELECT * FROM country_elections;

--
-- Arrays
CREATE FOREIGN TABLE main_exports (
    _id NAME,
    "mainExports" TEXT[]
) SERVER mongo_server OPTIONS (database 'mongo_fdw_regress', collection 'countries');
SELECT * FROM main_exports;
