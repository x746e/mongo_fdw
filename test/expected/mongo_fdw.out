CREATE SERVER mongo_server FOREIGN DATA WRAPPER mongo_fdw OPTIONS (address '127.0.0.1', port '27017');
\! mongoimport --db mongo_fdw_regress --collection countries --jsonArray --drop --quiet < test/sql/mongo_fixture.json
--
-- Basic types
CREATE FOREIGN TABLE countries (
    _id NAME,
    name VARCHAR,
    population INTEGER,
    capital VARCHAR,
    hdi FLOAT
) SERVER mongo_server OPTIONS (database 'mongo_fdw_regress', collection 'countries');
SELECT * FROM countries;
           _id            |  name   | population | capital  |  hdi  
--------------------------+---------+------------+----------+-------
 5381ccf9d6d81c8e8bf0434f | Ukraine |   45590000 | Kyiv     |  0.74
 5381ccf9d6d81c8e8bf04350 | Poland  |   38540000 | Warsaw   | 0.821
 5381ccf9d6d81c8e8bf04351 | Moldova |    3560000 | Chișinău |  0.66
(3 rows)

--
-- Subfields and dates
CREATE FOREIGN TABLE country_elections (
    _id NAME,
    "lastElections.type" VARCHAR,
    "lastElections.date" TIMESTAMP
) SERVER mongo_server OPTIONS (database 'mongo_fdw_regress', collection 'countries');
SELECT * FROM country_elections;
           _id            | lastElections.type |    lastElections.date    
--------------------------+--------------------+--------------------------
 5381ccf9d6d81c8e8bf0434f | presedential       | Sun May 25 00:00:00 2014
 5381ccf9d6d81c8e8bf04350 | parliamentary      | Sun Oct 09 00:00:00 2011
 5381ccf9d6d81c8e8bf04351 | parliamentary      | Sun Nov 28 00:00:00 2010
(3 rows)

--
-- Arrays
CREATE FOREIGN TABLE main_exports (
    _id NAME,
    "mainExports" TEXT[]
) SERVER mongo_server OPTIONS (database 'mongo_fdw_regress', collection 'countries');
SELECT * FROM main_exports;
           _id            |                                                                                         mainExports                                                                                         
--------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 5381ccf9d6d81c8e8bf0434f | {"Semi-finished products of iron or non-alloy steel","Flat-rolled products of iron or non-alloy steel","Sunflower-seed, safflower or cotton-seed oil"}
 5381ccf9d6d81c8e8bf04350 | {"Parts and accessories of the motor vehicles of headings 87.01 to 87.0","Motor cars and other motor vehicles principally designed for the transport","Reception apparatus for television"}
 5381ccf9d6d81c8e8bf04351 | {"Wine of fresh grapes, including fortified wines","Insulated (including enamelled or anodised) wire, cable","Sunflower seeds, whether or not broken"}
(3 rows)

